blaze-client/src/main/scala/org/http4s/client/blaze/Http1Connection.scala

```diff
@@ -107,9 +111,9 @@ private final class Http1Connection[F[_]](
   def resetRead(): Unit = {
     val state = stageState.get()
     val nextState = state match {
-      case Idle => Some(Idle)
+      case i: Idle => Some(i)
       case ReadWrite => Some(Write)
-      case Read => Some(Idle)
+      case Read => Some(Idle(Some(startIdleRead())))
```

Comment:
So the idle read will begin immediately after a connection is set to idle after being used ? Regardless of whether it is acquired again

Reply:
Exactly. This way we keep reading while the connection sits idle in the pool. If we happen to read an EOF, the connection gets closed. Once the connection is borrower and used, the read operation that we started here is used as the first operation in the process of reading a response.